src/http.ts

```diff
@@ -7,9 +7,19 @@ export const validatedFetch = async <T>(
   { decoding }: { decoding: "json" } = { decoding: "json" },
 ) => {
   const body = await (async () => {
+    const result = await response
+
+    if (result.status >= 400) {
+      throw new Error(
+        `Unexpected code: ${result.status}. Url: ${
+          result.url
+        }. Body: ${await result.text()}`,
+      )
+    }
+
     switch (decoding) {
       case "json": {
-        return (await response).json()
+        return result.json()
```

Comment:
wasn't it deferred to the consumer of this function?

i mean may be it would somewhere endup with 404, but it's fine (won't interrupt, so no need to try/catch), while now it will 100% throw error on 400+ ?

Reply:
Could you provide an example of how "it would somewhere endup with 404", without leading to `UnhandledRejection`? I feel like I'm missing something

And also, check the `await` on line 10

Reply:
No, that was me missing, i guess you're right it looks like it will always endup with error w/o try/catch. so the change looks good and more explicitly declaring form of error + trace.

I'm still curious - in the "Error before" statement, which is missing a trace, looks like coming from [Joi validation](https://github.com/paritytech/gitspiegel/blob/c7cca1ecddbb5277ba61fec910cbec3e44f1f651/src/configuration.ts#L172) --->> Do you think it's possible to make it, so all Joi validations globally would also include proper tracing?

Reply:
idk, need to check it out